File: java/src/test/java/problems201_300/ContainsDuplicateTest.java
package problems201_300;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ContainsDuplicateTest extends SampledTest {
    @Test
    public void test() {
        for (ContainsDuplicate solution : ContainsDuplicate.values()) {
            for (ContainsDuplicateSample sample : getSamples(ContainsDuplicateSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.containsDuplicate(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/ContainsDuplicateTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ContainsDuplicateTest : SampledTest() {
    @Test
    fun test() =
        ContainsDuplicate.entries.forEach { solution ->
            getSamples<Array<ContainsDuplicateSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.containsDuplicate(input.nums))
                    .isEqualTo(output)
            }
        }
}
